<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Data Atlas   for Behavior Geneticists   üåè</title>
    <meta charset="utf-8" />
    <meta name="author" content="S. Mason Garrison" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link href="libs/font-awesome/css/all.min.css" rel="stylesheet" />
    <link href="libs/font-awesome/css/v4-shims.min.css" rel="stylesheet" />
    <link rel="stylesheet" href="../slides.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

.title[
# Data Atlas <br> for Behavior Geneticists <br> üåè
]
.subtitle[
## Wake Forest University
]
.author[
### S. Mason Garrison
]

---





layout: true
  
&lt;div class="my-footer"&gt;
&lt;span&gt;
&lt;a href="https://DataScience4Psych.github.io/DataScience4Psych/" target="_blank"&gt;S. Mason Garrison&lt;/a&gt;
&lt;/span&gt;
&lt;/div&gt; 

---

&lt;!-- The adage 'A picture is worth a thousand words' is just as applicable to Behavior Genetics as it is to journalism. In this talk, I delve into the rich diverse history of data visualization techniques in the field of behavior genetics, tracing advancements from Wright's [@Wright1920] classic path diagram of guinea pigs to the contemporary usage of Manhattan plots. This survey sets the stage for the development of two resources aimed at bolstering the field's visualization toolkit. The first, an atlas employing 'ggplot2' in R, aims to guide researchers in selecting the most appropriate visualization techniques for their data. The second, a proposed R package, seeks to facilitate the integration of 'OpenMx'-compatible visualizations, streamlining the presentation of complex genetic analyses. Emphasizing practical advice, the talk will cover effective uses of path diagrams, family trees, and interaction visualizations, with the goal of improving clarity and interpretability of genetic data. This initiative aspires to democratize advanced data visualization tools, making them more accessible to the behavior genetics community and, in turn, amplifying the impact and reach of its research. --&gt;



---

class: middle

# Hello world!

&lt;!-- Slide 1: Introduction (1 minute) --&gt;
&lt;!-- Script: Good [morning/afternoon/evening], everyone. My name is S. Mason Garrison, and I'm from Wake Forest University. Today, I'm excited to talk about the importance of data visualization in the field of behavior genetics. As the saying goes, "A picture is worth a thousand words." This is particularly true in our field, where complex data can be better understood through effective visualization techniques. --&gt;

&lt;div class="figure" style="text-align: center"&gt;
&lt;img src="d00_slide_files/figure-html/unnamed-chunk-2-1.png" alt="QR code for these slides" width="30%" /&gt;
&lt;p class="caption"&gt;QR code for these slides&lt;/p&gt;
&lt;/div&gt;

.footnote[.center[
[r-computing-lab.github.io/slides/00_bga_2024/d00_slide](https://r-computing-lab.github.io/slides/00_bga_2024/d00_slide.html#1) 
]
]

---


# The Power of Visualization



### "A picture is worth a thousand words."

- Visualizations in behavior genetics:
  - Facilitate understanding of complex data.
  - Reveal patterns and relationships not obvious from raw data.
  - Enable more effective communication of research findings.


&lt;!-- Slide 2: The Power of Visualization (1 minute) --&gt;
&lt;!-- Script: The adage "A picture is worth a thousand words" is just as applicable to Behavior Genetics as it is to journalism. Visualizing data helps us to see patterns, relationships, and insights that might not be immediately apparent from raw numbers alone. Today, I will take you through the rich and diverse history of data visualization techniques in behavior genetics, from early methods to contemporary applications. --&gt;

---

class: middle

# Alternative Talk

- I *could* have spent this talk showing you some of the cool visualizations I've made since the last conference...

---

# Like These Ones!

.pull-left.center[
&lt;br&gt;
&lt;br&gt;
&lt;br&gt;
&lt;img src="img/X chroms.png" width="98%" style="display: block; margin: auto;" /&gt;
]

.pull-right[
&lt;img src="img/mtdna.png" width="75%" style="display: block; margin: auto;" /&gt;
]

---

# Or these!
.pull-left.center[
&lt;div class="figure" style="text-align: center"&gt;
&lt;img src="img/longevity.png" alt="A pedigree plot of the SSA mortality from 1900 to the present." width="99%" /&gt;
&lt;p class="caption"&gt;A pedigree plot of the SSA mortality from 1900 to the present.&lt;/p&gt;
&lt;/div&gt;
]
.pull-right[
&lt;div class="figure" style="text-align: center"&gt;
&lt;img src="img/sweden.png" alt="A pedigree plot of the Swedish mortality from 1750 to the present." width="99%" /&gt;
&lt;p class="caption"&gt;A pedigree plot of the Swedish mortality from 1750 to the present.&lt;/p&gt;
&lt;/div&gt;
]

---

# Or even my latest creation!


&lt;img src="img/upsidedownplots.png" width="40%" style="display: block; margin: auto;" /&gt;

.footnote[.center[
[x.com/SMasonGarrison/status/1800222792933310619](https://x.com/SMasonGarrison/status/1800222792933310619) 
]
]
--
.tiny[I swear that's these are extremely promising for making much better pedigrees plots...]


---

# Not today!


&lt;img src="img/nottoday.jpg" width="50%" style="display: block; margin: auto;" /&gt;

&lt;!--  Instead, I decided to focus some historic data visualizations in behavior genetics and gauge interest in a few resources I'm developing.
--&gt; 

---

class: middle
# Historical Context

---

## Wright's Path Diagram (1920)

- Sewall Wright's path diagrams were pioneering tools for illustrating genetic relationships.
- Enabled the visualization of complex genetic interactions and pathways.
- Revolutionized genetic research by providing a clear, visual method for understanding heritability and genetic correlations.

--

&lt;img src="img/Wright20aFigure5_1.jpg" width="30%" style="display: block; margin: auto;" /&gt;

&lt;!-- Slide 3: Historical Context (1 minute) --&gt;
&lt;!-- Script: One of the earliest and most significant contributions to data visualization in genetics was Wright's Path Diagram, introduced in 1920. This diagram allowed researchers to visually represent the relationships between genetic traits and their underlying factors, revolutionizing how we interpret genetic data. --&gt;


---

class: middle

# But that's... clich√©. 
-  Let's see some other examples from Sewall Wright's work.

--

&lt;img src="img/googlesearch.png" width="70%" style="display: block; margin: auto;" /&gt;

---

&lt;!-- Slide 4: Wright's Adaptive Landscape Diagram (2 minutes) 
This chapter traces the origins and conceptual lineages of the adaptive landscape concept and its representations. While Armand Janet's 1895 concept arguably marks the origin of the adaptive landscape concept and even its graphic representation, Janet's concept had very limited impact when compared to Sewall Wright's concept from 1932. As part of his effort to reconcile Mendelian genetics and Darwinian evolution in his shifting balance theory, Wright offered the metaphor of the adaptive landscape and its topographic representation as a way of depicting the effect of variations in population size, migration, and the strength of selection. Wright's genetic version of the adaptive landscape inspired other versions of the adaptive landscape based on phenotypic changes and on molecular changes. As a result, the history of the adaptive landscape is described in terms of three lineages based on the material basis of the adaptive landscape: the genetic landscape, the phenotypic landscape, and the molecular landscape.
--&gt;

## Sewall Wright's 1932 Adaptive Landscape Diagram


.pull-left[
- Wright's adaptive landscape diagram was an attempt to reconcile Mendelian genetics and Darwinian evolution. (Shifting Balance Theory)

&lt;img src="img/wright6panel.png" width="95%" style="display: block; margin: auto auto auto 0;" /&gt;
]

--

.pull-right[
- Offered a metaphor of the adaptive landscape and its topographic representation.
- Depicted the effect of variations in population size, migration, and the strength of selection.
- Other versions of the adaptive landscape were inspired by Wright's original diagram.
.small[- (Ok technically Armand Janet's conceptualized this in 1895, but Wright's concept had a much larger impact.)]
]



&lt;!-- Script: Wright's adaptive landscape diagram was another groundbreaking visualization, depicting the relationship between genetic variation and fitness. This provided a visual representation of natural selection and evolutionary dynamics. --&gt;

---

# Four adaptive landscape diagrams
.pull-left[
- Various adaptations and critiques of Wright's original diagram.
  - A: Ridley's depiction of a Fisherian landscape (1996).
  - B: Gavrilets‚Äô depiction of the holey landscape (1997).
  - C: Coyne et al.'s simplified Wrightian landscape (1997).
  - D: Kauffman and Levin's rugged adaptive landscape (1987).
]
&lt;br&gt;
.pull-right.center[
&lt;img src="img/responses towright.png" width="96%" style="display: block; margin: auto;" /&gt;
]
.footnote[Credit: Panel from Skipper (2004)]

&lt;!-- Four adaptive landscape diagrams, resulting from critiques of Wright's Reference Wright1932 diagram. Fig. 2a is Ridley's (Reference Ridley1996, 219) depiction of a Fisherian landscape. Fig. 2b is Gavrilets‚Äô (Reference Gavrilets1997, 309) depiction of the holey landscape. Fig. 2c is Coyne et al.'s (Reference Coyne, Barton and Turelli1997, 647) simplified Wrightian landscape. Fig. 2d is the Kauffman and Levin (Reference Kauffman and Levin1987, 33) rugged adaptive landscape. --&gt;

---

class: middle
# Fast Forward to the Present

---


## Modern Path Diagrams

- Modern path diagrams are used in structural equation modeling (SEM).
- Now... they use ‚óæs, ‚≠ïs, and ‚¨ÖÔ∏ès to represent variables, relationships, and causal pathways.
- Visualize complex relationships between variables in a statistical model in a way that can be easily interpreted.
- However, they can be challenging to create...

---

# .midi[Classic path diagrams can start to resemble ...]

&lt;!--  nuclear twin family (NTF) design --&gt;
.pull-left[
## Military Aircraft
&lt;img src="img/NTFD.path.diagram.png" width="90%" style="display: block; margin: auto;" /&gt;
.tiny[Keller, M. C., Medland, S. E., Duncan, L. E., Hatemi, P. K., Neale, M. C., Maes, H. H. M., &amp; Eaves, L. J. (2009). Modeling Extended Twin Family Data I: Description of the Cascade Model. Twin Research and Human Genetics, 12(1), 8‚Äì18. doi:10.1375/twin.12.1.8]
]

--
.pull-right[
## Storybook Characters
&lt;img src="img/Fig 5.jpg" width="90%" style="display: block; margin: auto;" /&gt;
.tiny[Hart, S.A., Little, C. &amp; van Bergen, E. Nurture might be nature: cautionary tales and proposed solutions. npj Sci. Learn. 6, 2 (2021). https://doi.org/10.1038/s41539-020-00079-z]
]

---

### Manhattan Plots

.pull-left-narrow[
- Widely used in genome-wide association studies (GWAS).
- Plot p-values for genetic variants across the genome.
- Help identify regions of the genome associated with traits or diseases.
- Visualize large-scale genetic data, making it easier to detect significant associations.
]
&lt;!-- Slide 4: Modern Techniques (1 minute) --&gt;
&lt;!-- Script: Moving to modern times, the Manhattan Plot has become a staple in genetic research, particularly for genome-wide association studies (GWAS). These plots help researchers identify significant genetic variants by visualizing p-values across the genome, making it easier to spot areas of interest. --&gt;
.footnote[All of Us Research Program Genomics Investigators (2024). Genomic data in the All of Us Research Program. Nature, 627(8003), 340‚Äì346. https://doi.org/10.1038/s41586-023-06957-x]

--

.pull-right-wide[
&lt;img src="img/Genomic data in the All of Us Research Program_manhattan.jpg" width="90%" style="display: block; margin: auto;" /&gt;
]

---

class: middle
# Importance of Good Graphs for Science Communication

---

## Case Study: Misinterpretation of Genetic Data
.pull-left[
- A figure in a high-profile Nature paper raised concerns about reinforcing racist beliefs.
- The UMAP generated figure aimed to show genetic diversity but was misinterpreted as supporting genetic essentialism.
- Highlights the need for careful design and interpretation of scientific figures.
.footnote.pull-left[- See more on the kerfuffle [here](https://www.nature.com/articles/d41586-024-00568-w)
- All of Us Research Program Genomics Investigators (2024). Genomic data in the All of Us Research Program. Nature, 627(8003), 340‚Äì346. https://doi.org/10.1038/s41586-023-06957-x]

]
--

.pull-right[
&lt;img src="img/Genomic data in the All of Us Research Program.jpg" width="70%" style="display: block; margin: auto;" /&gt;
]


&lt;!-- But a problem with using UMAP, Pritchard wrote, is that it can exaggerate the distinctiveness of populations and fail to represent their intermixing properly. --&gt;
---

## Lessons Learned
.pull-left[
- Good graphs should accurately represent data without misleading the audience.
- Consider the potential for misinterpretation and strive for clarity.
- Use visualizations to enhance understanding and communication of scientific findings.
&lt;img src="img/All of Us Research Program_A.jpg" width="60%" style="display: block; margin: auto;" /&gt;

]
--
.pull-right[
&lt;img src="img/Genomic data in the All of Us Research Program.jpg" width="70%" style="display: block; margin: auto;" /&gt;
]
&lt;!-- Slide 7: Importance of Good Graphs (2 minutes) --&gt;
&lt;!-- Script: This example underscores the importance of good graphs in science communication. A recent Nature article included a figure that was misinterpreted, leading to concerns about reinforcing racist beliefs. The chart aimed to show genetic diversity but ended up being misinterpreted as supporting genetic essentialism. This highlights the need for careful design and interpretation of scientific figures to ensure accurate and responsible communication of data. Good graphs should accurately represent data without misleading the audience, and researchers must consider the potential for misinterpretation. --&gt;

---

# Developing Resources

### Atlas with `ggplot2` in R

- A comprehensive guide for researchers to create effective visualizations using ggplot2.
- Includes examples of various types of plots and when to use them.
- Aims to standardize and improve the quality of visualizations in behavior genetics.
- Provides practical tips and best practices for creating clear and informative plots.
&lt;div class="figure" style="text-align: center"&gt;
&lt;img src="d00_slide_files/figure-html/unnamed-chunk-18-1.png" alt="QR code for the atlas" width="10%" /&gt;
&lt;p class="caption"&gt;QR code for the atlas&lt;/p&gt;
&lt;/div&gt;

.footnote[.center[
[r-computing-lab.github.io/data-atlas/](https://r-computing-lab.github.io/data-atlas/) 
]
]
&lt;!-- Slide 5: Developing Resources - Atlas (1 minute) --&gt;
&lt;!-- Script: To support researchers in choosing the most appropriate visualization techniques, we are developing an atlas using `ggplot2` in R. This resource aims to guide researchers through the process of selecting and implementing visualizations that enhance the clarity and interpretability of their data. --&gt;

---

## Proposed R Package

Integration with `OpenMx`:
- Streamlines the creation of visualizations for structural equation modeling (SEM) and other complex analyses. (Probably attempting to generalize `DiagrammeR` to works beyond RAM models)
- Provides functions for creating path diagrams, correlation matrices, and other visualizations compatible with the data structures used in `OpenMx`.
- Aims to make advanced visualization tools more accessible to researchers in behavior genetics.

&lt;!-- Slide 6: Developing Resources - R Package (1 minute) --&gt;
&lt;!-- Script: In addition to the atlas, I am gauging interest in developing an R package that integrates with `OpenMx`. This package will streamline the creation of visualizations for complex genetic analyses, making it easier for researchers to present their findings clearly and effectively. --&gt;

---

# Democratizing Visualization Tools

.pull-left[
### Making Advanced Scripts Accessible

- Open-source resources: Free and available to the entire research community.
- User-friendly packages: Simplify the process of creating high-quality visualizations.
- Educational materials: Provide guidance and training for researchers at all levels. Designed to complement what is already available from IBG and openMx and my own Data Science Materials.
]
--
.pull-right[

### Benefits:

- Improves the quality of visualizations in behavior genetics.
- Increases the impact and reach of research findings.
- Fosters collaboration and sharing of best practices.
- Amplify the reach and impact of behavior genetics research.
- Support the behavior genetics community with more tools and resources.
&lt;!-- Slide 8: Democratizing Visualization Tools (1 minute) --&gt;
&lt;!-- Script: One of our key goals is to democratize access to advanced visualization tools. By providing open-source resources and easy-to-use packages, we hope to make these powerful techniques accessible to the entire behavior genetics community, thus amplifying the impact of our research. --&gt;
]

---

class: middle

# Conclusion and Q&amp;A

## Advancing the Field through Visualization

- Today, we delved into some of the classic data visualizations in behavior genetics.
- We traced its evolution from Wright's path diagrams to modern Manhattan plots.
- We highlighted the development of new resources to enhance visualization practices in the field.
- Remember, effective visualization can transform complex data into understandable insights. Embrace these tools and improve your research communication.


Thank you for your attention.

&lt;!-- Slide 10: Conclusion (1 minute) --&gt;
&lt;!-- Script: In conclusion, data visualization plays a crucial role in advancing the field of behavior genetics. By enhancing our ability to communicate complex data clearly, we support greater collaboration and understanding within the scientific community. Thank you for your attention. --&gt;

---

## Any Questions?

Feel free to ask any questions now, or reach out to me after the talk via email _garrissm@wfu.edu_ or on github _github.com/smasongarrison_.

&lt;img src="d00_slide_files/figure-html/qr_ds4p-1.png" width="30%" style="display: block; margin: auto;" /&gt;

.footnote[.center[
[r-computing-lab.github.io/data-atlas/](https://r-computing-lab.github.io/data-atlas/) 
]
]
&lt;!-- Slide 11: Questions (1 minute) --&gt;
&lt;!-- Script: I‚Äôm happy to answer any questions you may have. Feel free to reach out to me via email or connect with me on GitHub. Thank you. --&gt;



    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"ratio": "16:9",
"highlightLines": true,
"highlightStyle": "solarized-light",
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
